server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -155,6 +159,122 @@ private static void internalParseDocument(MetadataFieldMapper[] metadataFieldsMa
         }
     }
 
+    private void addIgnoredSourceMissingValues(DocumentParserContext context) throws IOException {
+        Collection<IgnoredSourceFieldMapper.NameValue> ignoredFieldsMissingValues = context.getIgnoredFieldsMissingValues();
+        if (ignoredFieldsMissingValues.isEmpty()) {
```

Comment:
This would also be empty incase synthetic source isn't used?

Reply:
Correct, lemme assert.